pyproject.toml

```diff
@@ -27,3 +27,75 @@ pytest-benchmark = "^4.0.0"
 [build-system]
 requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
+[tool.ruff]
+target-version = "py39"
+
+[tool.ruff.lint]
+select = [
+  "AIR",    # Airflow
+  "ASYNC",  # flake8-async
+  "BLE",    # flake8-blind-except
+  "C4",     # flake8-comprehensions
+  "C90",    # McCabe cyclomatic complexity
+  "DJ",     # flake8-django
+  "E",      # pycodestyle
+  "F",      # Pyflakes
+  "FURB",   # refurb
+  "G",      # flake8-logging-format
+  "ICN",    # flake8-import-conventions
+  "INT",    # flake8-gettext
+  "ISC",    # flake8-implicit-str-concat
+  "LOG",    # flake8-logging
+  "NPY",    # NumPy-specific rules
+  "PD",     # pandas-vet
+  "PERF",   # Perflint
+  "PT",     # flake8-pytest-style
+  "Q",      # flake8-quotes
+  "S",      # flake8-bandit
+  "SLF",    # flake8-self
+  "SLOT",   # flake8-slots
+  "T10",    # flake8-debugger
+  "TCH",    # flake8-type-checking
+  "TID",    # flake8-tidy-imports
+  "W",      # pycodestyle
+  "YTT",    # flake8-2020
```

Comment:
Why enable these specifically. Some of these e.g. `"AIR"`, and `"DJ"` will never apply to this code base.

Reply:
Ruff is written in Rust, not Python so it is so fast that it can lint the entire CPython codebase in 1/3 of a second.  If a test passes, include it even if you think this codebase will never need it.  Codebases have a habit of morphing in unpredictable ways.  There is almost no runtime cost to putting up a guardrail but there can be a cost if a guardrail is not in place.

Adding `FLY` and `UP` was #20.  I use `ruff check --select=ALL --statistics | sort -k2` to find the failing tests.  Then I disable only the failing tests -- if a test passes, keep it. Then in PRs to follow, fix a few codes until you can run `ruff check --select=ALL` with a small number of `--ignore` or `per-file-ignores`.